libparsec/crates/platform_storage/src/workspace.rs

```diff
@@ -217,20 +224,35 @@ impl WorkspaceStorage {
     }
 
     pub async fn set_prevent_sync_pattern(&mut self, pattern: &Regex) -> anyhow::Result<bool> {
```

Comment:
```suggestion
    pub async fn set_prevent_sync_pattern(&mut self, pattern: Regex) -> anyhow::Result<bool> {
```
My own rule of thumb on passing by reference vs by ownership:
- if the ownership is always needed in the body of the function (which is the case here, cf `self.prevent_sync_pattern = pattern.clone();`) then it should be passed by ownership.
- if the ownership is needed in corner case (typically when building errors), then passing by reference likely the good approach

Reply:
done